WordPress/gutenberg
# Add filter to preload paths

Another go at https://github.com/WordPress/gutenberg/issues/10268 since my previous PR no longer merged cleanly.

## Description
Preload paths is an array of local API paths that are preloaded. The response data for these paths are populated in PHP and then the preload middleware ensures that the data is returned immediately when fetched, without needing to make an external request. 

This PR adds a filter to preload paths, which means developers can add additional paths. 

See https://github.com/WordPress/gutenberg/issues/10268

## How has this been tested?

* Add the following code in PHP (e.g. in your theme functions.php file)

```
add_filter( 'block_editor_preload_paths', function( $paths, $post ) {
	if ( $post->post_type === 'page' ) {
		$paths[] = '/wp/v2/types/post?context=edit';
	}
	return $paths;
}, 10, 2 );
```

* Then create a new Page. (Our filter only works on post type page!) 
* Open developer tools and from the console enter the following: 

```
wp.apiFetch({ path: '/wp/v2/types/post?context=edit' });
```

* You will see that the returned value is a promise that is already resolved. 
* If you remove the filter you added in PHP, and retry, the response is a promise that is pending. When it does resolve, the data will be the same. 
 
## Screenshots 
NA

## Types of changes





## Checklist:
